shared/processing/schemaGetters.ts

```diff
@@ -0,0 +1,354 @@
+    ISchemas,
+    ISchemaString,
+    IWrappedItem,
+    ZapiSchemas,
+} from '../types/dataModels';
+import {
+    getFromLinkedItems,
+    processLinkedItemsElement,
+    processMultipleChoiceElement,
+    processTaxonomyElement,
+} from '../utils/processElements';
+import {
+    getSystemProperties,
+    getWrappedData,
+    processItemsInto,
+} from './common';
+import RichTextField = Fields.RichTextField;
+
+export const getSchemaDataFromLinkedItemElement = (
+    items: ZapiSchemas[],
+    dataBlob: IPreprocessedData,
+    linkedItems: ContentItem[],
+): Array<IWrappedItem<ISchemas>> => {
+    const schemas = [];
+
+    items.map((schema: ZapiSchemas) => {
+        schemas.push(getSchemaData(schema, dataBlob, linkedItems));
+    });
+
+    return schemas;
+};
+
+export const getSchemaDataFromRichTextElement = (
+    field: RichTextField,
+    dataBlob: IPreprocessedData,
+    linkedItems: ContentItem[],
+): Array<IWrappedItem<ISchemas>> => {
+    const schemas = [];
+
+    field.linkedItemCodenames.map((codename: string) => {
+        const schema = getFromLinkedItems<ZapiSchemas>(codename, linkedItems);
+
+        schemas.push(getSchemaData(schema, dataBlob, linkedItems));
+    });
+
+    return schemas;
+};
+
+const getSchemaData = (
+    schema: ZapiSchemas,
+    dataBlob: IPreprocessedData,
+    linkedItems: ContentItem[],
+): IWrappedItem<ISchemas> | undefined => {
+    switch (schema.system.type) {
+        case 'zapi_schema__array': {
+            processItemsInto<ISchemas>(getSchemaDataFromRichTextElement)(schema.items, dataBlob, linkedItems);
+
+            return getSchemaArrayData(schema as ZapiSchemaArray);
+        }
+        case 'zapi_schema__anyof': {
+            processItemsInto<ISchemas>(getSchemaDataFromLinkedItemElement)(schema.schemas, dataBlob, linkedItems);
+
+            return getSchemaAnyOfData(schema as ZapiSchemaAnyof);
+        }
+        case 'zapi_schema__allof': {
+            processItemsInto<ISchemas>(getSchemaDataFromRichTextElement)(schema.schemas, dataBlob, linkedItems);
+
+            return getSchemaAllOfData(schema as ZapiSchemaAllof);
+        }
+        case 'zapi_schema__boolean': {
+            return getSchemaBooleanData(schema as ZapiSchemaBoolean);
+        }
+        case 'zapi_schema__integer': {
+            return getSchemaIntegerData(schema as ZapiSchemaInteger);
+        }
+        case 'zapi_schema__number': {
+            return getSchemaNumberData(schema as ZapiSchemaNumber);
+        }
+        case 'zapi_schema__object': {
+            processItemsInto<IPropertyReferencingASchema>(getPropertyReferencingASchemaData)(
+                schema.properties,
+                dataBlob,
+                linkedItems,
+            );
+            processItemsInto<ISchemas>(getSchemaDataFromRichTextElement)(
+                schema.properties,
+                dataBlob,
+                linkedItems,
+            );
+            processItemsInto<ISchemas>(getSchemaDataFromLinkedItemElement)(
```

Comment:
Same case as already mentioned. But consider looking at it from a "dependency injection" point of view... These pre-initialized method might be shared across modules...